clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -1341,24 +1328,25 @@ trueDualPortBlockRamModel !_clkA enA weA addrA datA !_clkB enB weB addrB datB =
     deepErrorX ("Write enabled, but address unknown; position " <> show n <>
                 "\nAddress error message: " <> msg)
 
-  getConflict :: Bool -> Int -> Bool -> Int -> Maybe Conflict
-  getConflict enableA addrA_ enableB addrB_ =
+  getConflict :: Bool -> Bool -> Bool -> Int -> Bool -> Int -> Maybe Conflict
+  getConflict enA_ enB_ wenA addrA_ wenB addrB_ =
     -- If port A or port B is writing on (potentially!) the same address,
     -- there's a conflict
-    if sameAddr then maybeConflict else Nothing
+    if sameAddr then Just conflict else Nothing
```

Comment:
Having `Maybe Conflict` instead of just `Conflict` is now an optimization step, right? That is, `Conflict` could signal there is no conflict by virtue of having all its fields set to `False`.

To be clear, I'm not opposed to this, I actually like it. I'm asking for my own understanding.

Reply:
I guess it is an "optimization" step. Or at least, `Nothing` indicates "no conflict".